packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -116,11 +119,13 @@ export abstract class QueryBuilderState implements CommandRegistrar {
       functionsExplorerState: observable,
       fetchStructureState: observable,
       filterState: observable,
+      watermarkState: observable,
       resultState: observable,
       textEditorState: observable,
       unsupportedQueryState: observable,
       showFunctionsExplorerPanel: observable,
       showParametersPanel: observable,
+      isEditingWatermark: observable,
```

Comment:
let's put these in `watermarkState` to keep `QueryBuilderState` clean

Reply:
+ 1